crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2669,43 +2666,92 @@ impl<'db> KnownInstanceType<'db> {
         self.class().to_instance(db)
     }
 
-    pub fn try_from_module_and_symbol(module: &Module, instance_name: &str) -> Option<Self> {
-        if !module.search_path().is_standard_library() {
-            return None;
-        }
-        match (module.name().as_str(), instance_name) {
-            ("typing", "Any") => Some(Self::Any),
-            ("typing", "ClassVar") => Some(Self::ClassVar),
-            ("typing", "Deque") => Some(Self::Deque),
-            ("typing", "List") => Some(Self::List),
-            ("typing", "Dict") => Some(Self::Dict),
-            ("typing", "DefaultDict") => Some(Self::DefaultDict),
-            ("typing", "Set") => Some(Self::Set),
-            ("typing", "FrozenSet") => Some(Self::FrozenSet),
-            ("typing", "Counter") => Some(Self::Counter),
-            ("typing", "ChainMap") => Some(Self::ChainMap),
-            ("typing", "OrderedDict") => Some(Self::OrderedDict),
-            ("typing", "Optional") => Some(Self::Optional),
-            ("typing", "Union") => Some(Self::Union),
-            ("typing", "NoReturn") => Some(Self::NoReturn),
-            ("typing", "Tuple") => Some(Self::Tuple),
-            ("typing", "Type") => Some(Self::Type),
-            ("typing", "Callable") => Some(Self::Callable),
-            ("typing" | "typing_extensions", "Annotated") => Some(Self::Annotated),
-            ("typing" | "typing_extensions", "Literal") => Some(Self::Literal),
-            ("typing" | "typing_extensions", "LiteralString") => Some(Self::LiteralString),
-            ("typing" | "typing_extensions", "Never") => Some(Self::Never),
-            ("typing" | "typing_extensions", "Self") => Some(Self::TypingSelf),
-            ("typing" | "typing_extensions", "Final") => Some(Self::Final),
-            ("typing" | "typing_extensions", "Concatenate") => Some(Self::Concatenate),
-            ("typing" | "typing_extensions", "Unpack") => Some(Self::Unpack),
-            ("typing" | "typing_extensions", "Required") => Some(Self::Required),
-            ("typing" | "typing_extensions", "NotRequired") => Some(Self::NotRequired),
-            ("typing" | "typing_extensions", "TypeAlias") => Some(Self::TypeAlias),
-            ("typing" | "typing_extensions", "TypeGuard") => Some(Self::TypeGuard),
-            ("typing" | "typing_extensions", "TypeIs") => Some(Self::TypeIs),
-            ("typing" | "typing_extensions", "ReadOnly") => Some(Self::ReadOnly),
-            _ => None,
+    pub fn try_from_file_and_symbol(
```

Comment:
Oh, I see we already had the `*_and_symbol` method naming pattern here, where "symbol" meant "name"... that doesn't really change my opinion of it, though :) It's not wrong, I think in the context of our codebase it's just unnecessarily confusing.